### PR TITLE
feat(decorators): select, dispatch, dispatchAll

### PR DESCRIPTION
**Pull request by kosz**
Connected to #40

### CODE REVIEW

File: examples/counter/pages/App.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  <counter></counter>
+  `
+})
+
+export class App {
+
+    @select(state => state.counter) funcCounter$;
+    @select('counter') stringKey$;
+    @select(state => state.counter * 2) counterX2$: Observable<any>;
+    foo: any;
```
Comment (SethDavenport) [inline examples/counter/pages/App.ts]: wat

**Comment by SethDavenport** [comment]
But yeah this is looking nice.  Users of ngRedux can decide where they want to put their dispatches this way.

:shipit:

File: examples/counter/pages/App.ts (hunk 1 above)
Comment (kosz) [inline examples/counter/pages/App.ts]: yea I need to add the markup in the template that uses those.

File: src/components/ng-redux.ts [hunk 2]
```diff
@@ -14,8 +14,10 @@ import {
 
 import { Observable } from 'rxjs/Observable';
 import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+
 import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/distinctUntilChanged';
+import 'rxjs/add/operator/combineLatest';
```
Comment (e-schultz) [inline src/components/ng-redux.ts]: I'd leave this out of ng-redux - and leave it upto the consumers to impor tthe operators that they need. internally - we need distinctUntilChanged - but not map and combineLatest

**Comment by SethDavenport** [comment]
re-:shipit:

**Pull request merged**